src/core/unit.h

```diff
@@ -381,6 +381,9 @@ typedef struct UnitStatusMessageFormats {
         const char *starting_stopping[2];
         const char *finished_start_job[_JOB_RESULT_MAX];
         const char *finished_stop_job[_JOB_RESULT_MAX];
+        /* If an entry in finished_{start,stop}_job is NULL and this field isn't,
+         * it'll be called to provide a context-dependent format string, or NULL to fall back to generic. */
```

Comment:
This feels inverted: I would expect the following hierarchy: finished_job(), the finish_start_job and finish_stop_jobs fields, the generic fallback, or in other words, to have finished_job() with the highest priority, and then fall back to previous logic without any changes.

Reply:
I've reworded the comment and moved `finished_job()` above the original  handler.